pkg/rhcos/ami_regions.go

```diff
@@ -0,0 +1,33 @@
+package rhcos
+
+import (
+	"context"
+
+	"github.com/coreos/stream-metadata-go/arch"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// AMIRegions returns the AWS regions in which an RHCOS AMI for the specified architecture is published.
+func AMIRegions(architecture types.Architecture) sets.String {
+	stream, err := FetchCoreOSBuild(context.Background())
+	if err != nil {
+		logrus.Error("could not fetch the rhcos stream data: %w", err)
+		return nil
```

Comment:
Why does `nil` need to be handled by the caller? A `nil` return means that no regions were found. Ultimately, the error is a result of previous code failing to validate the architecture appropriately. The only reason I am printing an error at all is for the benefit of a future developer knowing that they forgot to do something when adding a new architecture.

Reply:
If you return nil rather than  an empty set, won't this cause an exception here: https://github.com/openshift/installer/pull/5466/files#diff-bab1b8072f369df53e27d65607145fb93caabd3a86fdefbcafb176e823726069R83?

Reply:
No.

Reply:
> The only reason I am printing an error at all is for the benefit of a future developer knowing that they forgot to do something when adding a new architecture.

Also, this explanation makes sense, so if we expect this to never be nil except when there's a programmers error I'm fine with letting it cause an exception (not handle the `nil`)

Reply:
> If you return nil rather than an empty set, won't this cause an exception here: https://github.com/openshift/installer/pull/5466/files#diff-bab1b8072f369df53e27d65607145fb93caabd3a86fdefbcafb176e823726069R83?

Apparently it doesn't cause an exception. Not sure I understand why not... to me it appears to be calling `Has()` on nil but testing here seems ok: https://go.dev/play/p/aikW4rf-4TJ

Reply:
Ah ok. I see now sets.String is a map https://github.com/kubernetes/apimachinery/blob/master/pkg/util/sets/string.go#L27

Reply:
Also, note that in go it is valid to call functions on a nil object. The receiver of the function is ultimately just a parameter passed to the function